### PR TITLE
Add Firewall Settings Endpoints, Tests

### LINKED ISSUE #87: Add ability to change zone level security level
https://api.cloudflare.com/#zone-settings-change-security-level-setting

The only related fields I can find for this are related to Page Rules, there does not appear to be an option to change the overall zone security level.

### PR DESCRIPTION
**Pull request by exportsmedia**
Closes #87

### CODE REVIEW

File: src/Endpoints/FirewallSettings.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            'zones/' . $zoneID . '/settings/security_level',
+            [
+                'value' => $value,
+            ]
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->success) && $body->success == true) {
+            return true;
+        }
+        return false;
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: Consider following code:

```php
return isset($body->success) && $body->success == true;
```

File: src/Endpoints/FirewallSettings.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    public function updateChallengeTTLSetting(string $zoneID, int $value)
+    {
+        $return = $this->adapter->patch(
+            'zones/' . $zoneID . '/settings/challenge_ttl',
+            [
+                'value' => $value,
+            ]
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->success) && $body->success == true) {
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: It has the same suggested change I mention on previous comment.

File: src/Endpoints/FirewallSettings.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    public function updateBrowserIntegrityCheckSetting(string $zoneID, string $value)
+    {
+        $return = $this->adapter->patch(
+            'zones/' . $zoneID . '/settings/browser_check',
+            [
+                'value' => $value,
+            ]
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->success) && $body->success == true) {
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: It has the same suggested change I mention on previous comment.

File: src/Endpoints/FirewallSettings.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+     * @param string $zoneID The ID of the zone
+     * @return string|false
+     */
+    public function getSecurityLevelSetting(string $zoneID)
+    {
+        $return = $this->adapter->get(
+            'zones/' . $zoneID . '/settings/security_level'
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->result)) {
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: As the package requires `php-7.0` version at least, try considering following code snippets:

```php
return $body->result->value ?? false;
```

File: src/Endpoints/FirewallSettings.php [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+     * @param string $zoneID The ID of the zone
+     * @return integer|false
+     */
+    public function getChallengeTTLSetting(string $zoneID)
+    {
+        $return = $this->adapter->get(
+            'zones/' . $zoneID . '/settings/challenge_ttl'
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->result)) {
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: It has the same suggested changes I mention on previous comment.

File: src/Endpoints/FirewallSettings.php [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+     * @param string $zoneID The ID of the zone
+     * @return string|false
+     */
+    public function getBrowserIntegrityCheckSetting(string $zoneID)
+    {
+        $return = $this->adapter->get(
+            'zones/' . $zoneID . '/settings/browser_check'
+        );
+        $body = json_decode($return->getBody());
+        if (isset($body->result)) {
```
Comment (peter279k) [inline src/Endpoints/FirewallSettings.php]: It has the same suggested changes I mention on previous comment.

File: src/Endpoints/FirewallSettings.php (hunk 2 above)
Comment (IcyApril) [reply src/Endpoints/FirewallSettings.php]: Personally prefer the syntax used as it is a bit more explicit and readable.

Comment (peter279k) [reply src/Endpoints/FirewallSettings.php]: @IcyApril, thank you for your reply.

That's fine  if these syntax is readable and consistency.

Go ahead with this original syntax.

**Pull request merged**